src/components/title-blog-index/title-blog-index.jsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState, useRef, useCallback } from "react";
+
+import { ArrowIndexIcon } from '../icons';
+import { SeeMoreTitles } from "../see-more-titles/see-more-titles";
+
+import { classnames, findTitles } from "../../helpers";
+
+import * as styles from "./title-blog-index.module.scss";
+
+const TitleBlogIndex = ({ data, refIndexTitles, disappearIndex }) => {
+  const [titleOnView, setTitleOnView] = useState('');
+  const [selectLink, setSelectLink] = useState('');
+  const [isTopArrow, setIsTopArrow] = useState(false);
+  const [isBottomArrow, setIsBottomArrow] = useState(false);
+  const subContainerRef = useRef(null);
+  const viewContainerRef = useRef(null);
+
+  useEffect(() => {
+    window.addEventListener('scroll', getActiveTitle, { passive: true });
+
+    checkOverflow();
+    const handleResize = () => checkOverflow();
+    window.addEventListener('resize', handleResize);
+
+    return () => {
+      window.removeEventListener('scroll', getActiveTitle);
+      window.removeEventListener('resize', handleResize);
+    };
+  }, []);
+
+  const getActiveTitle = () => {
+    if (refIndexTitles.current) {
+      const elementList = Array.from(refIndexTitles.current.childNodes);
+      const titles = findTitles(elementList);
+      const scrollPosition = window.scrollY;
+      let activeTitle = titles[0];
+      titles.forEach(title => {
+        const titleTop = title.getBoundingClientRect().top + window.scrollY - 500;
+        if (scrollPosition >= titleTop) {
+          activeTitle = title;
+        }
+      });
+      setTitleOnView(activeTitle?.textContent);
+      setSelectLink('');
+    }
+  };
```

Comment:
What do you do here?

Reply:
With this I can mark the index I am currently in, if it matches the title, then the index will change.

https://github.com/xmartlabs/xl-blog/assets/107500715/6212180c-a660-4605-a12b-0249486f404c

Reply:
Ok, I think this can change as simpler:
```suggestion
  const getActiveTitle = () => {
    if (refIndexTitles.current) {
      const elementList = Array.from(refIndexTitles.current.childNodes);
      const titles = findTitles(elementList);
      const scrollPosition = window.scrollY;
      const activeTitle = titles.findLast((title) => title.getBoundingClientRect().top <= 500) || titles[0];
      setTitleOnView(activeTitle.textContent);
      setSelectLink('');
    }
  };
```
What do you think?